aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/CredentialCaches.kt

```diff
@@ -191,6 +191,48 @@ class CredentialCacheConfig(codegenContext: ClientCodegenContext) : ConfigCustom
                 }
             }
 
+            ServiceConfig.RuntimePluginConfig("layer") -> {
```

Comment:
Could all this logic be computed once during ServiceRuntimePlugin construction rather than per request?

I'm currently working on a PR that makes the ServiceRuntimePlugin constructed once per client rather than per request.

Reply:
Sorry, I think I misunderstood where the `ServiceConfig.RuntimePluginConfig` hook was emitting code into. I was thinking it was going into the `ServiceRuntimeConfig::config` method, but reading it closer, it looks like it goes somewhere into the service config. So I think it's fine the way it is.

Note that matching on `ServiceConfig.RuntimePluginConfig("layer")` is probably not what you want though:
```suggestion
            is ServiceConfig.RuntimePluginConfig -> {
```

Reply:
No worries, thanks for asking this question. Updated the section name (95318ea) and fixed the matching (9cc52cf).